### PR TITLE
Fix build, use layout API and conform to Haiku coding style.

### PR DESCRIPTION
**Pull request by AkshayAgarwal007**

### CODE REVIEW

**Comment by AkshayAgarwal007** [comment]
bedead

File: src/DeathWindow.cpp [hunk 1]
```diff
@@ -1,96 +1,128 @@
...
 #include <OS.h>
-#include <Beep.h>
+#include <Roster.h>
 
-#include "DeathWindow.h"
 
+DeathWindow::DeathWindow()
+	:
+	BWindow(BRect(), "BeDead", B_TITLED_WINDOW,
+			 B_AUTO_UPDATE_SIZE_LIMITS | B_WILL_ACCEPT_FIRST_CLICK)
```
Comment (humdingerb) [inline src/DeathWindow.cpp]: Only one indent.

File: src/DeathWindow.cpp [hunk 2]
```diff
@@ -1,96 +1,128 @@
...
 
-const uint32 KILL_KILL_KILL = 'KiLL';
-int32 PIDIndex[10000];
+	fBackView = new BView("BackView", B_WILL_DRAW);
+	rgb_color bgcolor = ui_color(B_PANEL_BACKGROUND_COLOR);
+	fBackView->SetViewColor(bgcolor);
 
-DeathWindow::DeathWindow() : BWindow(BRect(100,100,400,400),"BeDead",B_TITLED_WINDOW,B_NOT_RESIZABLE | B_NOT_ZOOMABLE | B_WILL_ACCEPT_FIRST_CLICK) {
+	fDeathList = new BListView("Death List", B_SINGLE_SELECTION_LIST,
+					B_WILL_DRAW);
```
Comment (humdingerb) [inline src/DeathWindow.cpp]: Only one indent.

File: src/DeathWindow.cpp [hunk 3]
```diff
@@ -1,96 +1,128 @@
...
 
-	DeathList = new BListView(BRect(0,0,285,250),"Death List",B_SINGLE_SELECTION_LIST);
+	fDeathScroll = new BScrollView("scroll_death", fDeathList,							B_WILL_DRAW, false, true);
```
Comment (humdingerb) [inline src/DeathWindow.cpp]: Newline + one indent.

File: src/DeathWindow.cpp [hunk 4]
```diff
@@ -1,96 +1,128 @@
...
 		
-	DeathList->AttachedToWindow();
-	CloseCheck->AttachedToWindow();
-
-	AddChild(DeathScroll);
-	AddChild(CloseCheck);
-
-	AddApps(DeathList);
+	fDeathList->AttachedToWindow();
+	fCloseCheck->AttachedToWindow();
```
Comment (humdingerb) [inline src/DeathWindow.cpp]: I've never seen the hook AttachedToWindow() being explicitely called. Is this really necessary?

File: src/DeathWindow.cpp [hunk 5]
```diff
@@ -1,96 +1,128 @@
...
+DeathWindow::MessageReceived(BMessage *message)
+{
 	switch(message->what) {
 		case B_SOME_APP_LAUNCHED:
-			DeathList->MakeEmpty();
-			AddApps(DeathList);		
+			fDeathList->MakeEmpty();
+			AddApps(fDeathList);
 			break;
+
```
Comment (humdingerb) [inline src/DeathWindow.cpp]: cases need extra { } as there are more than one instructions in the case blocks.

File: src/DeathWindow.cpp [hunk 6]
```diff
@@ -1,96 +1,128 @@
...
 			break;
-		case KILL_KILL_KILL:
-			int32 index;
-			message->FindInt32("index",&index);
-			KillApp(DeathList,index);
-			DeathList->MakeEmpty();
-			AddApps(DeathList);	
+
+		case kAppKill:
+		{	int32 index;
```
Comment (humdingerb) [inline src/DeathWindow.cpp]: { on a separate line.

File: src/DeathWindow.cpp [hunk 7]
```diff
@@ -1,96 +1,128 @@
...
-void DeathWindow::KillApp(BListView *DeathList,int32 index) {
-	kill_team(PIDIndex[index]);
-	if(CloseCheck->Value()) be_app->PostMessage(B_QUIT_REQUESTED);	
-}
\ No newline at end of file
+void
+DeathWindow::KillApp(BListView* DeathList, int32 index)
+{
+	kill_team(fPIDIndex[index]);
+	if(fCloseCheck->Value())
```
Comment (humdingerb) [inline src/DeathWindow.cpp]: space after "if".

**Review by humdingerb — CHANGES_REQUESTED** [review]
If you're serious with the Haiku coding style guide... :)

**Comment by AkshayAgarwal007** [comment]
Updated...
Despite how much I try to avoid them either you or @waddlesplash always catch some coding style mistakes in my PRs :)

**Comment by AkshayAgarwal007** [comment]
It's marked as freeware, so what should be included in the license headers?

**Comment by humdingerb** [comment]
I guess you can try to contact the author at retroj@iserv.net, but I doubt the address will work after 18 years. The website isn't there anymore anyway...

@waddlesplash , what to do in such a case? Use MIT until we hear back otherwise?

**Comment by waddlesplash** [comment]
Don't put any license and just have the copyright dates and original author.

**Comment by AkshayAgarwal007** [comment]
I've added the Copyright headers.

**Comment by scottmc** [comment]
@waddlesplash, any idea where this came from and whether permission was given by the original author to post it here?  How do we know it is "open source"?  Did the original author send it to anyone via email with permission and if so what license did they agree too?  These are key things we should be asking of every program we post here.

**Comment by clasqm** [comment]
@scottmc I  am the one who uploaded it, and no, I am not the original author. I agree that it is something we need to think about these days, and at least if we have it, there is a chance of finding the original author and asking them. But these programs come from a more innocent time, when we were not yet all obsessed with licensing.

**Pull request merged**